File: 020/020.py
memoize = {}

def factorial(input):
    if memoize.has_key(input):
        return memoize[input]
    if input == 0: return 1
    result = memoize[input] = factorial(input - 1) * input
    return result

print sum(int(a) for a in str(factorial(100)))

File: 020/020.rb
@memoize = Hash.new

def fact(n)
    return @memoize[n] if @memoize.has_key? n
    return 1 if n == 0
    @memoize[n] = result = n * fact(n - 1)
    return result
end
puts fact(100).to_s.bytes.inject(0) { |n, t| t + n - ?0 }

